### ISSUE TITLE
function-component-definition should ignore function with `this`

### ISSUE THREAD

**Issue by wmertens** [yannickcr/eslint-plugin-react#2781]
This is a polyfill for IE:

```js
		Element.prototype.closest = function (s) {
			let el = this
			if (!document.documentElement.contains(el)) return null
			do {
				if (el.matches(s)) return el
				el = el.parentElement || el.parentNode
			} while (el !== null && el.nodeType === 1)
			return null
		}
```

it needs access to `this` and function-component-definition complains about it.

**Comment by ljharb** [comment]
I'd say that, in addition, a function assigned to a prototype method shouldn't be considered a component.

Of course, modifying browser globals, and using `=` to install a polyfill, and running a react linter on that polyfill, all don't make any sense, but that's neither here nor there.

**Comment by himanshu007-creator** [comment]
i would like to work on this issue
